test.js

```diff
@@ -209,3 +209,28 @@ test('fallback to cwd if `module.filename` is `null`', t => {
 	module.filename = preservedFilename;
 	del.sync(conf.path, {force: true});
 });
+
+test('onDidChange()', t => {
+	let runs = 0;
+	const check = (newValue, oldValue) => {
+		runs++;
+		t.is(newValue, '🐴');
+		t.is(oldValue, fixture);
+	};
+
+	t.context.conf.set('foo', fixture);
+	const unsubscribeFoo = t.context.conf.onDidChange('foo', check);
+	t.context.conf.set('foo', '🐴');
+	unsubscribeFoo();
+	t.context.conf.set('foo', fixture);
+
+	t.context.conf.set('baz.boo', fixture);
+	const unsubscribeBaz = t.context.conf.onDidChange('baz.boo', check);
+	t.context.conf.set('baz.boo', '🐴');
+	unsubscribeBaz();
+	t.context.conf.set('baz.boo', fixture);
+
+	if (runs !== 2) {
+		t.fail();
+	}
+});
```

Comment:
Can you add a test with using `.delete()` to ensure the newValue is correct then (`undefined`), and then using `.set()` to ensure the old value is `undefined`.